app/models/foreman_teamdynamix/host_extensions.rb

```diff
@@ -2,18 +2,18 @@ module ForemanTeamdynamix
   module HostExtensions
     extend ActiveSupport::Concern
 
-    def td_api
-      @td_api ||= TeamdynamixApi.instance
-    end
-
     included do
-      before_create :create_teamdynamix_asset
+      include Orchestration::Teamdynamix
       before_destroy :retire_teamdynamix_asset
       validates :teamdynamix_asset_uid, uniqueness: { :allow_blank => true }
     end
 
     private
 
+    def td_api
+      @td_api ||= TeamdynamixApi.instance
+    end
+
```

Comment:
We actually can;t move this method under private. Things outside this model need to access that method. You can put it below included though. Just above private